### ISSUE TITLE
LR Schedule Hyperparameters

### ISSUE THREAD

**Issue by Zasder3** [openai/CLIP#107]
Hello, I'm trying to replicate your LR scheduling as described within the paper and have a couple of questions:

1. Is your scheduler the one with or without restarts, i.e. is it `torch.optim.lr_scheduler.CosineAnnealingWarmRestarts`?
2. When you specify warmup iterations, is that the period (`T_0`) of the cosine function or is that something different altogether?

Thank you!

**Comment by jongwook** [comment]
1. The LR schedule didn't have restarts, the multiplier becomes (almost) 1.0 after warmup, and monotonically decreases 0.0 drawing a cosine curve over 32 epochs.
2. A linear warmup was done over the first 2,000 iterations and is not dependent of the period of the cosine function.
